### PR TITLE
CORE: View a list of comments associated with an article

### PR DESCRIPTION
**Pull request by tonymm55**
Added functionality to show comments for a particular article_id. Chose to use a button for this on the SingleArticleView component. Button click appeared to do nothing, but comments were below the page view, so added scrollIntoView.

### CODE REVIEW

File: src/Components/Articles.jsx [hunk 1]
```diff
@@ -6,24 +6,33 @@ import ArticleList from "./ArticlesList.jsx";
...
         setArticles(reponse.data.articles);
+        setLoading(false);
       });
     } else {
       fetchArticles(selectedTopic).then((response) => {
         setArticles(response.data.articles);
+        setLoading(false);
       });
     }
   }, [selectedTopic]);
```
Comment (nc-christian-dutton) [inline src/Components/Articles.jsx:24]: You're definitely on the right track here, but if you route by :topic then you can use useParams to grab if there's a topic or not and render this component on multiple routes. Then, you can just call fetchArticles once without the if condition, always passing it a topic. In your axios call you can add a second argument for params, there's an example in the notes. I think this is working anyway but could be tidied up a little. I'll be doing something like this in the seminar tomorrow!

**Comment by nc-christian-dutton** [comment]
Looking good! i'd remove your console logs but otherwise good work 👍

File: src/Components/SingleArticleView.jsx [hunk 2]
```diff
@@ -1,48 +1,80 @@
...
           console.error("Failed to fetch article:", error);
           setLoading(false);
         });
+
+      fetchCommentsByArticleId(articleId)
+        .then((response) => {
+          if (response && response.data) {
+            setComments(response.data.comments);
+          } else {
+            console.error("No data in response");
```
Comment (nc-christian-dutton) [inline src/Components/SingleArticleView.jsx:37]: You can use this else block to give the user some info, article 37 I believe has no comments on it yet, would you want an error in that case or just display nothing? Maybe let the user know!

File: src/Components/Articles.jsx (hunk 1 above)
Comment (tonymm55) [reply src/Components/Articles.jsx:24]: Ooh, nice will look forward to that seminar to get further understanding!  Thanks for the feedback Christian 😊

File: src/Components/SingleArticleView.jsx (hunk 2 above)
Comment (tonymm55) [reply src/Components/SingleArticleView.jsx:37]: Good point! Yes, I would want a popup/alert to say no comments present or similar!

File: src/Components/Articles.jsx (hunk 1 above)
Comment (tonymm55) [reply src/Components/Articles.jsx:24]: That seminar was really useful and makes the comment a lot clearer to me now 😊

**Pull request merged**